### PR TITLE
Rename bibtexkey

### PR DESCRIPTION
**Pull request by calixtus**
As discussed in the devcall, BibtexKey should now be renamed to CitationKey

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/cli/JabRefCLI.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ private static Options getOptions() {
         options.addOption("h", "help", false, Localization.lang("Display help on command line options"));
         options.addOption("n", "nogui", false, Localization.lang("No GUI. Only process command line options"));
         options.addOption("asfl", "automaticallySetFileLinks", false, Localization.lang("Automatically set file links"));
-        options.addOption("g", "generateBibtexKeys", false, Localization.lang("Regenerate all keys for the entries in a BibTeX file"));
+        options.addOption("g", "generateCitationKeys", false, Localization.lang("Regenerate all keys for the entries in a BibTeX file"));
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/cli/JabRefCLI.java:157]: Please add a changelog entry (especially for the changed cli option)

File: src/main/java/org/jabref/gui/keyboard/KeyBinding.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
 public enum KeyBinding {
 
     ABBREVIATE("Abbreviate", Localization.lang("Abbreviate journal names"), "ctrl+alt+A", KeyBindingCategory.TOOLS),
-    AUTOGENERATE_BIBTEX_KEYS("Autogenerate BibTeX keys", Localization.lang("Autogenerate BibTeX keys"), "ctrl+G", KeyBindingCategory.QUALITY),
+    AUTOGENERATE_BIBTEX_KEYS("Autogenerate citation keys", Localization.lang("Autogenerate citation keys"), "ctrl+G", KeyBindingCategory.QUALITY),
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/keyboard/KeyBinding.java:8]: The enum keys still have `Bibtex` in their name

**Comment by calixtus** [comment]
Tobias 'EagleEye' Diez 😉 Thanks.

**Comment by koppor** [comment]
Why "Citation key"? Because BibLaTeX calls it like that:

grafik

**Pull request merged**